### DISCUSSION TITLE
PIC32 and ESP32 without Arduino

### DISCUSSION THREAD

**Discussion by JorgenSandberg** [Bodmer/TFT_eSPI#2520] [New processor/display support or capability request]
The TFT_eSPI library is excellent. I am wondering whether there is a simple way to make it work on a PIC32 processor and on ESP32 without Arduino i.e. on an IDE with Visual Studio Code and the ESP-IDF version 5 only.

I have noticed that the TFT_eSPI.h file includes <Arduino.h>, <Print.h> and <SPI.h> that are not part of the TFT_eSPI repository. Is it possible to do without the three?

The generic processor defined in the Processors folder makes use of some Arduino native functions for parallel operation of the TFT, and some SPI functions. Would it be possible to make it even more generic by dropping the <Arduino.h>, <Print.h> and <SPI.h> header files and define the functionality of a few functions that the user would have to supply for the interface between the library and his processor?

**Comment by Bodmer** [comment]
The library requires the Arduino environment support files. I don't plan to remove this requirement.

**Comment by JorgenSandberg** [comment]
Dear Mr Bodmer,

 

I understand. I will look for something else then.

Thank you for your reply.

 

Regards,

Jorgen

**Reply by wavejaco** [reply]
Hi Jorgen

Did you find an alternative or other means of running this library on a PIC32?

**Reply by JorgenSandberg** [reply]
Dear wavejaco,
Thank you for your interest.
I managed to get some of the TFT_eSPI to run on my ESP32 without the Arduino environment using the Espressif IDE (hence no Arduino, no Visual Studio Code, no PlatformIO). I made some modifications of the code. However, I prefer a graphic package born independent of Arduino and written in C or C++, so I continued my search.
Currently, I am trying the LVGL. It is running on my PIC32MZ processor now using the MPLAB X and the XC32 compiler. The next step is to port it to my ESP32-S3 board using the Espressif IDE. The LVGL is written in pure C, and it is well documented making it relatively easy to port to any processor as long as the compiler handles C.
Regards,
Jorgen

**Reply by wavejaco** [reply]
Dear Jorgen

Many thanks for your reply.

Your setup with PIC32MZ, MPLAB X and XC32 is exactly the combination I am looking for, except that I am using a PIC32MX. I am not familiar with LVGL, but will have a good look at it. A quick search and read on it seems like what I am looking for.

I have started an attempt to port the TFT_eSPI library to C for the PIC32MX, but it very soon became obvious that this won't be a trivial exercise. So thus I might rather be looking at alternatives, like LVGL.

I will have a good look at LVGL and give it a try. Might be just what I am looking for.

Best regards
Jaco
